database/migrations/2016_07_14_154357_MysqlToPostgres.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+use Symfony\Component\Console\Output\ConsoleOutput;
+
+class MysqlToPostgres extends Migration
+{
+    private $console;
+
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        $this->console = new ConsoleOutput();
+
+        // Generate pgloader config
+        $mysqlConnection = "from mysql://" . env('DB_USERNAME') . ":" . env('DB_PASSWORD') . "@" . env('DB_HOST') . "/" . env('DB_DATABASE');
+        $postgresConnection = "into postgresql://" . env('POSTGRESQL_DB_USERNAME', 'homestead') . ":" . env('POSTGRESQL_DB_PASSWORD', 'secret') . "@" . env('POSTGRESQL_DB_HOST', 'localhost') . "/" . env('POSTGRESQL_DB_DATABASE', 'homestead');
+
+        $header = "LOAD DATABASE";
+        $body = <<<'EOD'
+with truncate
+
+CAST type datetime to timestamp drop default drop not null using zero-dates-to-null,
+      type date drop not null drop default using zero-dates-to-null
+
+EXCLUDING TABLE NAMES MATCHING 'migrations';
+EOD;
+
+        $output = implode("\n", array($header, $mysqlConnection, $postgresConnection, $body));
+        $configPath = base_path() . "/pfmimport.load";
+        file_put_contents($configPath, $output);
+
+        // Run pgloader
+        $this->execRunWithCallback("pgloader " . $configPath);
```

Comment:
Pgloader can be installed from apt
sudo apt-get install pgloader

Make sure version 3 is installed by running pgloader --version

On Sat, 16 Jul 2016, 10:48 Peter Deltchev, <notifications@github.com> wrote:

> In database/migrations/2016_07_14_154357_MysqlToPostgres.php
> <https://github.com/Poniverse/Pony.fm/pull/96#discussion_r71063948>:
>
> > +        $header = "LOAD DATABASE";
> > +        $body = <<<'EOD'
> > +with truncate
> > +
> > +CAST type datetime to timestamp drop default drop not null using zero-dates-to-null,
> > +      type date drop not null drop default using zero-dates-to-null
> > +
> > +EXCLUDING TABLE NAMES MATCHING 'migrations';
> > +EOD;
> > +
> > +        $output = implode("\n", array($header, $mysqlConnection, $postgresConnection, $body));
> > +        $configPath = base_path() . "/pfmimport.load";
> > +        file_put_contents($configPath, $output);
> > +
> > +        // Run pgloader
> > +        $this->execRunWithCallback("pgloader " . $configPath);
>
> This generated the following message on my dev environment:
> sh: 1: pgloader: not found
>
> The migration was marked as successful but it looks like pgloader needs
> to be installed or otherwise configured in the Vagrant setup scripts
> (Ansible as well?).
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/Poniverse/Pony.fm/pull/96/files/06afc0c9d7e8e57019b6b271262e96548d890da2#r71063948>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAiYvnukJ5lvMm8Nr6gDbv590jVs9bgUks5qWKjpgaJpZM4JLPiF>
> .
>